components/notify/notify.js

```diff
@@ -10,15 +11,30 @@ module.exports = class Notify {
     this.notifyTemplate = options.notifyTemplate;
   }
 
-  send(email) {
+  async sendEmail(templateId, recipient, personalisation) {
     const reference = uuidv4();
 
-    return this.notifyClient.sendEmail(this.notifyTemplate, email.recipient, {
-      personalisation: {
-        'email-subject': email.subject,
-        'email-body': email.body
-      },
-      reference });
+    try {
+      await this.notifyClient.sendEmail(templateId, recipient, {
+        personalisation: personalisation,
+        reference });
+    } catch (error) {
+      console.error('Error sending email:', error.response ? error.response.data : error.message);
+    }
+  }
+
+  send(email) {
+    let personalisation = {
+      'email-subject': email.subject,
+      'email-body': email.body
+    };
+
+    if (email.attachment && email.attachment !== undefined) {
+      personalisation = Object.assign({'email-attachment':
+        this.notifyClient.prepareUpload(email.attachment)}, personalisation);
+    }
+
+    this.sendEmail(this.notifyTemplate, email.recipient, personalisation);
```

Comment:
I think it would be good to make this function async as well and to use a try/catch block

Reply:
I have changed it to async